src/server.cpp

```diff
@@ -1054,10 +1056,15 @@ void CServer::DecodeReceiveData ( const int iChanCnt, const int iNumClients )
                     emit ClientDisconnected ( iCurChanID ); // TODO do this outside the mutex lock?
                 }
 
+                FreeChannel ( iCurChanID ); // note that the channel is now not in use
```

Comment:
I didn't think so. I thought it would be better if the channel still existed at the point the jam controller was notified. But in fact the channel object continues to exist - `FreeChannel()` just removes it from the channel order list and the binary search.

But I admit I haven't yet tested this code with recording enabled. It would be good to do so.

Reply:
Well I've run a test server with recording enabled, and some test clients. All appears to work ok. I've also looked at the code for `CJamRecorder::OnDisconnected()`, and there doesn't appear to be anything that would interact with the `server.cpp` code.

Reply:
I'm okay if it makes no difference.